packages/nextjs-template/components/DendronSearch.tsx

```diff
@@ -49,33 +53,84 @@ function DebouncedDendronSearchComponent(props: DendronPageWithNoteDataProps) {
   // gets fresh results.
   const debouncedSearch: SearchFunction | undefined = fuse
     ? _.debounce<SearchFunction>((query, setResults) => {
-        setResults(fuse.search(query));
+        if (_.isUndefined(query)) {
+          return;
+        }
+        setResults(fuse.search(query.substring(1)));
       }, SEARCH_DELAY)
     : undefined;
   return (
     <DendronSearchComponent {...props} {...results} search={debouncedSearch} />
   );
 }
 
+type SearchResults = Fuse.FuseResult<NoteProps>[] | undefined;
+
 type SearchProps = Omit<ReturnType<typeof useFetchFuse>, "fuse"> & {
   search: SearchFunction | undefined;
 };
 
 function DendronSearchComponent(
   props: DendronPageWithNoteDataProps & SearchProps
 ) {
+  const { noteIndex, dendronRouter, search, error, loading, notes } = props;
+
+  const [searchResults, setSearchResults] =
+    React.useState<SearchResults>(undefined);
+  const [lookupResults, setLookupResults] = React.useState<NoteIndexProps[]>(
+    []
+  );
+  const [results, setResults] =
+    React.useState<"searchResults" | "lookupResults">();
   const dispatch = useCombinedDispatch();
   const { noteBodies, requestNotes } = useNoteBodies();
-  const { noteIndex, dendronRouter, search, error, loading, ensureIndexReady } =
-    props;
-  const [value, setValue] = React.useState("");
-  const [results, setResults] = React.useState<SearchResults>(undefined);
+  const lookup = useDendronLookup();
+  const { noteActive } = useNoteActive(dendronRouter.getActiveNoteId());
+  const initValue = noteActive?.fname || "";
+  const [value, setValue] = React.useState(initValue);
+
   useEffect(() => {
-    if (search) search(value, setResults);
+    if (search) {
+      search(value, setSearchResults);
+    }
   }, [value, search]);
+
+  useEffect(() => {
+    requestNotes(searchResults?.map(({ item: note }) => note.id) || []);
+  }, [requestNotes, searchResults]);
```

Comment:
why do we need to depend on `requestNotes` function here?

Reply:
Good question,

React doesn't know exactly when the requestNotes can change, in this case, is just a way to make sure nothing will change in the requestNotes function.

This is the kind of warnings thrown by the linter, that BTW is turned off.

This is a [good blogpost](https://reacttraining.com/blog/when-to-use-functions-in-hooks-dependency-array/) explaining exactly this behaviour.

Reply:
got it. thanks for the explanation. feel free to re-enable the linting rules in that case in a separate PR!

just for my understanding, in this case, since `requestNotes` doesn't close over state, this technically isn't necessary but the linter doesn't know that and this could change in the future?